### PR TITLE
Add way to configure rhsso/rhbk via settings

### PR DESCRIPTION
**Pull request by mdujava**
- onboards on 3scale-api 0.35.0
- allows to specify rhsso.kind to be rhsso or rhbk
- fix problem with not respecting fixtures.tools.namespace for sso lookup
- add --sso-only flag to skip non sso tests

### CODE REVIEW

File: testsuite/dynaconf_loader.py [hunk 1]
```diff
@@ -109,24 +109,33 @@ def _apicast_image(ocp):
...
+                return tools.secrets["credential-sso"]["ADMIN_PASSWORD"].decode("utf-8")
         except (OpenShiftPythonException, KeyError):
             try:
                 # try 7.4 known deployment if previous fails
                 return tools.environ("dc/sso")["SSO_ADMIN_PASSWORD"]
             except (OpenShiftPythonException, KeyError):
-                return None
+                pass
+
+    return None
```
Comment (mijaros) [inline testsuite/dynaconf_loader.py:138]: This depth seems kinda redundant and deep, every positive branch ends with the return so nesting the code into the except branches is not good idea to be honest.

File: testsuite/tests/conftest.py [hunk 2]
```diff
@@ -183,6 +184,27 @@ def pytest_collection_modifyitems(session, config, items):
...
+        to_remove = True
+        for fixture in selected_fixtures:
+            if fixture in item.fixturenames:
+                to_remove = False
+
+        if to_remove:
+            selected.remove(item)
+            deselected.append(item)
+
+    items[:] = selected
```
Comment (mijaros) [inline testsuite/tests/conftest.py:205]: I want on the record that I don't like this! :-)

**Review by mijaros — CHANGES_REQUESTED** [review]
I think that the changes are well written and i understand that the nesting is relict of older development, but why don't fix it now?

**Review by mijaros — APPROVED** [review]
Ok, this looks very promissing.

File: testsuite/tests/conftest.py (hunk 2 above)
Comment (dhlavac) [inline testsuite/tests/conftest.py:205]: Cannot be simplified to ?
`   selected_fixtures = {"rhsso_setup"}
    deselected = []

    sso_only_opt = config.option.sso_only
    selected = []

    for item in items:
        for marker in item.iter_markers(name="issue"):
            issue = marker.args[0]
            issue_id = issue.rstrip("/").split("/")[-1]
            item.user_properties.append(("issue", issue))
            item.user_properties.append(("issue-id", issue_id))

        if sso_only_opt and not any(fixture in selected_fixtures for fixture in item.fixturenames):
            deselected.append(item)
        else:
            selected.append(item)

    items[:] = selected

    if deselected:
        config.hook.pytest_deselected(items=deselected)`

**Review by JaurbanRH — DISMISSED** [review]

File: Pipfile [hunk 3]
```diff
@@ -60,6 +60,8 @@ pytest-persistence = "*"
 pytest-metadata = ">=3.0.0"
 pytest-html = "*"
 ansi2html = "*"
+ipdb = "*"
```
Comment (mkudlej) [inline Pipfile:63]: This should e part of different PR.

**Review by mkudlej — APPROVED** [review]
LGTM except of one change in Pipfile, which should be removed I think.

**Review by mkudlej — APPROVED** [review]

File: testsuite/tests/conftest.py [hunk 4]
```diff
@@ -176,13 +177,28 @@ def pytest_collection_modifyitems(session, config, items):
...
+    deselected = []
+
     for item in items:
         for marker in item.iter_markers(name="issue"):
             issue = marker.args[0]
             issue_id = issue.rstrip("/").split("/")[-1]
             item.user_properties.append(("issue", issue))
             item.user_properties.append(("issue-id", issue_id))
 
+        if sso_only_opt and not any(fixture in selected_fixtures for fixture in item.fixturenames):
```
Comment (dhlavac) [inline testsuite/tests/conftest.py:193]: we need to deselect performance tests from sso tag, there are 4 tests from performance

**Review by dhlavac — APPROVED** [review]

**Pull request merged**